src/main/java/org/killbill/billing/plugin/adyen/core/KillbillAdyenNotificationHandler.java

```diff
@@ -204,6 +203,32 @@ private void handleNotification(final NotificationRequestItem item) {
         handleNotification(null, item);
     }
 
+    private void handleChargebackNotification(final NotificationRequestItem item, boolean success) {
+        final NotificationItem notification = new NotificationItem(item);
+        final DateTime utcNow = clock.getUTCNow();
+
+        final AdyenResponsesRecord originalRecord = getResponseRecord(item.getOriginalReference());
+        final UUID kbAccountId = UUID.fromString(originalRecord.getKbAccountId());
+        final UUID kbTenantId = UUID.fromString(originalRecord.getKbTenantId());
+        final UUID kbPaymentId = UUID.fromString(originalRecord.getKbPaymentId());
+
+        final CallContext context = new AdyenCallContext(utcNow, kbTenantId);
+        final Account account = getAccount(kbAccountId, context);
+
+        UUID kbPaymentTransactionId = null;
+        try {
+            final Payment payment = osgiKillbillAPI.getPaymentApi().createChargeback(account, kbPaymentId, notification.getAmount(), Currency.valueOf(notification.getCurrency()), null, context);
```

Comment:
>  I can't use the pspReference as it is possible that we receive more than one chargeback for one payment and then it is possible that the pspReference will be used more than once.

Adyen doesn't give us any unique reference for that particular chargeback? Populating a random UUID doesn't help here for tracking (Kill Bill will do it for you if you pass null anyways).